### PR TITLE
[terra-application-navtigation] Theme updates

### PR DESCRIPTION
**Pull request by dkasper-was-taken**
### Summary
Structural and variable updates to address clinical theme needs.

Thanks for contributing to Terra.
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-application-navigation/src/common/_PopupCount.jsx [hunk 1]
```diff
@@ -8,13 +8,17 @@ import styles from './PopupCount.module.scss';
 const cx = classNames.bind(styles);
 
 const propTypes = {
+  /**
+   * Whether or not the count should be used as placeholder height.
```
Comment (emilyrohrbough) [inline packages/terra-application-navigation/src/common/_PopupCount.jsx]: this description is a little confusing. Is this prop for toggling height or for hiding/showing the count value?

File: packages/terra-application-navigation/src/terra-dev-site/test/application-navigation/HeroAndUser.test.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import Placeholder from 'terra-doc-template/lib/Placeholder';
+
+const hero = <Placeholder title="Test Hero" variant="light" />;
+
+const userConfig = {
+  name: 'Test Name',
+  detail: 'Test Detail',
+};
+
+// TODO: remove terra-application after it is incorporated into dev-site
```
Comment (emilyrohrbough) [inline packages/terra-application-navigation/src/terra-dev-site/test/application-navigation/HeroAndUser.test.jsx]: I this this TODO has been resolved: https://github.com/cerner/terra-dev-site/blob/master/package.json#L110

File: packages/terra-application-navigation/src/common/_PopupMenuListItem.jsx [hunk 3]
```diff
@@ -117,7 +117,7 @@ const PopupMenuListItem = ({
         ) : null}
       {icon ? <div className={cx('icon')}>{icon}</div> : null}
       <div className={cx('text')}>{text}</div>
-      {notificationCount > 0 && <PopupCount value={notificationCount} isInline className={cx('extension-row-count')} />}
+      {<PopupCount isPlaceholder={!notificationCount} value={notificationCount || 0} isInline className={cx('extension-row-count')} />}
```
Comment (emilyrohrbough) [inline packages/terra-application-navigation/src/common/_PopupMenuListItem.jsx]: Which examples /wdio screenshots demonstrate's this change?

Comment (dkasper-was-taken) [reply packages/terra-application-navigation/src/common/_PopupMenuListItem.jsx]: Extensions, utils, and tab rollup should all have a larger default row height now.

File: packages/terra-application-navigation/src/terra-dev-site/test/application-navigation/HeroAndUser.test.jsx (hunk 2 above)
Comment (dkasper-was-taken) [reply packages/terra-application-navigation/src/terra-dev-site/test/application-navigation/HeroAndUser.test.jsx]: Updated.

File: packages/terra-application-navigation/src/common/_PopupCount.jsx [hunk 4]
```diff
@@ -8,13 +8,18 @@ import styles from './PopupCount.module.scss';
 const cx = classNames.bind(styles);
 
 const propTypes = {
+  /**
+   * Whether or not the count should be present for height calculations, but hidden from view.
+   * To ensure that row heights are consistent, we need the ability to have the count and not see it.
+   */
+  isPlaceholder: PropTypes.bool,
```
Comment (emilyrohrbough) [inline packages/terra-application-navigation/src/common/_PopupCount.jsx]: Would it make sense to rename to `isHidden`

File: packages/terra-application-navigation/CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,7 @@ Unreleased
 * Cleaned up imports in examples and test files
 * Added check to prevent rendering of Utility Button or Nav Drawer Button if they are empty.
 * Updated `keycode-js` from `v1.0.4` to `v2.0.1`
+* Update underlying structure to handle new theme variables.
```
Comment (StephenEsser) [inline packages/terra-application-navigation/CHANGELOG.md]: This snuck into a released section during a branch update

Comment (dkasper-was-taken) [reply packages/terra-application-navigation/CHANGELOG.md]: Updated.

File: packages/terra-application-navigation/src/common/_PopupCount.jsx (hunk 4 above)
Comment (dkasper-was-taken) [reply packages/terra-application-navigation/src/common/_PopupCount.jsx]: Updated.

**Pull request merged**